xls/modules/rle/rle.x

```diff
@@ -0,0 +1,465 @@
+            TestEncOutData {symbol: TestSymbol:0xC, count: TestCount:0x3, last:false},
+            TestEncOutData {symbol: TestSymbol:0xC, count: TestCount:0x3, last:false},
+            TestEncOutData {symbol: TestSymbol:0x3, count: TestCount:0x3, last:false},
+            TestEncOutData {symbol: TestSymbol:0x2, count: TestCount:0x2, last:true},
+        ];
+        let (tok, _, _) = for ((counter, data_out), (tok, can_repeat, prev_symbol)): ((u32, TestEncOutData) , (token, bool, TestSymbol)) in enumerate(trans_recv) {
+            let (tok, enc_output) = recv(tok, enc_output_r);
+            let _ = trace_fmt!(
+                "Received {} transactions, symbol: 0x{:x}, count: {}, last: {}",
+                counter, enc_output.symbol, enc_output.count, enc_output.last
+            );
+            let _ = assert_eq(enc_output, data_out);
+            let max_count = std::unsigned_max_value<TEST_COUNT_WIDTH>();
+
+            let _ = assert_eq(enc_output.count > TestCount:0, true);
+            let _ = assert_eq(enc_output.count <= max_count, true);
+            let data_repeats = prev_symbol == enc_output.symbol;
+            let _ = assert_eq(data_repeats && can_repeat, data_repeats);
+            ((tok, (enc_output.count == max_count)|| enc_output.last, enc_output.symbol))
+        }((tok, true, TestSymbol:0));
+
+        // `last` after `last` check
+        let trans_send: TestEncInData[2] = [
+            TestEncInData {symbol: TestSymbol:0x1, last: true},
+            TestEncInData {symbol: TestSymbol:0x2, last: true},
+        ];
+        let tok = for ((counter, data_in), tok): ((u32, TestEncInData) , token) in enumerate(trans_send) {
+            let tok = send(tok, enc_input_s, data_in);
+            let _ = trace_fmt!("Sent {} transactions, symbol: 0x{:x}, last: {}",
+                counter, data_in.symbol, data_in.last);
+            (tok)
+        }(tok);
+        let trans_recv: TestEncOutData[2] = [
+            TestEncOutData {symbol: TestSymbol:0x1, count: TestCount:0x1, last:true},
+            TestEncOutData {symbol: TestSymbol:0x2, count: TestCount:0x1, last:true},
+        ];
+        let (tok, _, _) = for ((counter, data_out), (tok, can_repeat, prev_symbol)): ((u32, TestEncOutData) , (token, bool, TestSymbol)) in enumerate(trans_recv) {
+            let (tok, enc_output) = recv(tok, enc_output_r);
+            let _ = trace_fmt!(
+                "Received {} transactions, symbol: 0x{:x}, count: {}, last: {}",
+                counter, enc_output.symbol, enc_output.count, enc_output.last
+            );
+            let _ = assert_eq(enc_output, data_out);
+            let max_count = std::unsigned_max_value<TEST_COUNT_WIDTH>();
+
+            let _ = assert_eq(enc_output.count > TestCount:0, true);
+            let _ = assert_eq(enc_output.count <= max_count, true);
+            let data_repeats = prev_symbol == enc_output.symbol;
+            let _ = assert_eq(data_repeats && can_repeat, data_repeats);
```

Comment:
would that be triggered if we were using the symbol here? or do we need a separate test?